pkg/controller/dataimportcron-controller.go

```diff
@@ -242,10 +255,20 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 	}
 
 	desiredDigest := dataImportCron.Annotations[AnnSourceDesiredDigest]
-	if desiredDigest != "" && (imports == nil || (importSucceeded && desiredDigest != imports[0].Digest)) {
-		if err := r.createImportDataVolume(ctx, dataImportCron); err != nil {
-			return res, err
+	digestUpdated := desiredDigest != "" && (len(imports) == 0 || desiredDigest != imports[0].Digest)
+	if digestUpdated {
```

Comment:
There are 4 permutations here, in the unit tests below I see only 2 tests. Can we add some unit tests that ensure that the corner cases are doing what they are supposed to do?